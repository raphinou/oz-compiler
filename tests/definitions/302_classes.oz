% Default values for method arguments
local
   C I
in
   class C
      meth init()
         skip
      end
      meth hello(name:Name<='Doe' firstname:Firstname)
         {Show Name#Firstname}
      end
      meth greet(Name<='Unknown')
         {Show 'Hello '#Name}
      end
   end
   I = {New C init()}
   {I hello(firstname:'Alan' name:'Turing')}
   {I hello(firstname:'Jon')}
   {I greet('Mozart')}
   {I greet()}
end
