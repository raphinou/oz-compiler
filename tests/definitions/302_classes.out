'Turing'#'Alan'
'Doe'#'Jon'
'Hello '#'Mozart'
'Hello '#'Unknown'
